pkg/logger/operationid_middleware.go

```diff
@@ -17,9 +20,11 @@ const OpIDHeader OperationIDKey = "X-Operation-ID"
 func OperationIDMiddleware(handler http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		ctx := WithOpID(r.Context())
```

Comment:
it looks like the middleware will always generate its own operation ID. Is there a reason why it shouldn't try to read the value from the `X-Operation-ID` header to facilitate the correlation between CS and Maestro?

I'm also surprised that Maestro doesn't seem to log the operation ID. IIUC it's only reported to the client in the HTTP response when an error occurs:

https://github.com/openshift-online/maestro/blob/adb7e9012bd01b12c10bcbb86eaf2e0bfa0126f9/pkg/handlers/framework.go#L32-L41